types/lodash/common/common.d.ts

```diff
@@ -213,6 +213,7 @@ declare module "../index" {
     type NotVoid = unknown;
     type IterateeShorthand<T> = PropertyName | [PropertyName, any] | PartialShallow<T>;
     type ArrayIterator<T, TResult> = (value: T, index: number, collection: T[]) => TResult;
+    type TupleIterator<T extends readonly unknown[], TResult> = (value: T[number], index: { [K in keyof T]: StringToNumber<K> }[never], collection: T) => TResult;
```

Comment:
For the index type, does `index: StringToNumber<keyof T>` work?

Reply:
Thanks, it works like a charm.